mobile-save-for-later/src/main/scala/com/gu/sfl/identity/IdentityService.scala

```diff
@@ -36,19 +36,35 @@ class IdentityServiceImpl(identityConfig: IdentityConfig, okHttpClient: OkHttpCl
       .get()
       .build()
 
+    def extractUserIdFromSuccessResult(response: Response) = {
+      Try {
+        val body = response.body().string()
+        logger.debug(s"Identity api response: $body")
+        val node = mapper.readTree(body.getBytes)
+        node.path("id").textValue
+      } match {
+        case Success(userId) =>
+          // .textValue can return null, so wrap in Option.apply
+          promise.success(Option(userId))
+        case Failure(_) =>
+          promise.success(None)
+      }
+    }
+
     okHttpClient.newCall(
       request
     ).enqueue(new Callback {
       override def onResponse(call: Call, response: Response): Unit = {
-        Try {
-           val body = response.body().string()
-           logger.debug(s"Identity api response: $body")
-           val node = mapper.readTree(body.getBytes)
-           node.get("user").path("id").textValue
-         } match {
-           case Success(userId) => promise.success(Some(userId))
-           case Failure(_)  => promise.success(None)
-         }
+        val responseCodeGroup = response.code() / 100
```

Comment:
Does this get cast to an `Int` ? as `503/100 = 5.03`

Reply:
It performs integer division. `response.code()` returns an Int.

Reply:
Cool